com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/IDebugAdapterContext.java

```diff
@@ -27,6 +29,17 @@
      */
     void sendEvent(Events.DebugEvent event);
 
+    /**
+     * Send a request to the DA.
+     * @param request
+     *              the request.
+     * @param timeout
+     *              the timeout (in millis).
+     * @param cb
+     *              the request call back function.
+     */
+    void sendRequest(Messages.Request request, int timeout, Consumer<Messages.Response> cb);
```

Comment:
If we have the IProtocolServer instance, we don't need to expose sendRequest or sendEvent through the IDebugAdapterContext interface. A context itself is more natural for saving references instead of providing functionalities.